### PR TITLE
Optimize SVG icons

### LINKED ISSUE #4076: Lots of SVG source files broken since svgo optimization
When wanting to edit some icons, I stumbled upon this issue:
capture du 2017-03-26 18-57-33
capture du 2017-03-26 18-55-42

This seems to be caused by the SVGO optimization: https://github.com/nextcloud/server/pull/1534

While they seem to be correctly displayed, Inkscape has big problems showing them correctly. There’s a reason we use Scour as per https://github.com/nextcloud/server/blob/master/core/img/image-optimization.sh#L10 and not SVGO. ;)

cc @skjnldsv could you check it out? In doubt I’d say your PR needs to be reverted and we should do it via Scour instead – which was already used on most of the icons.

### PR DESCRIPTION
**Pull request by pixelipo**
Ready for review. This is first part of the #4076 (`actions` icon types).

The whole to-do list has been moved to a more generic issue #5157

### CODE REVIEW

**Comment by pixelipo** [comment]
I've stumbled upon an icon called `pause-big.svg` - it's the same as `pause.svg`, but 40% larger (within the same 16px viewBox). Anyone know where (and why) this icon is used? Can we drop it? @nextcloud/designers

**Comment by jancborchardt** [comment]
It's used in the Gallery app detail view / slideshow, bottom right.
But yeah, now that we use SVG only (that was a remnant of PNG times) we could simply do it by CSS.

**Comment by pixelipo** [comment]
Thanks @jancborchardt - I'll add it to the Todo list or create a new issue to be fixed after merging this

**Comment by MorrisJobke** [comment]
@pixelipo Is it possible to make smaller PRs out of this? Otherwise we would end up with a diff that is quite big.

**Comment by pixelipo** [comment]
@MorrisJobke diff is kind of useless for testing this PR.

I have finished fixing `actions` icons so I can stop at this point and create new PRs for other icon types. Changes so far are only those required to make sure that the code is valid and the icons fit the predefined requirements (16*16 size) plus some "weight-loss" (the whole folder is down to 39kB from ~76kB).

The only ways to test this are:
1. Open icon in Inkscape to make sure it is not "scrambled"
2. check if there are any broken icons in various components and apps

I can cut it into several smaller PRs you think it will help - but tell me how much files per PR?

**Comment by MorrisJobke** [comment]
> I can cut it into several smaller PRs you think it will help - but tell me how much files per PR?

I guess the amount in here is enough. Checking already ~100 icons could be hard. And that's the reason I asked to split it. Dumping more stuff into it would not make it easier to review. ;)

**Comment by jancborchardt** [comment]
Good stuff - @nextcloud/designers let's test and review this! :) Also make sure to open in your SVG editor of choice (e.g. Inkscape) to check they are not mangled anymore.

File: core/img/actions/settings-white.svg [hunk 1]
```diff
@@ -1 +1 @@
...
+<svg xmlns="http://www.w3.org/2000/svg" version="1.1" height="16" width="16" viewBox="0 0 16 16"><path opacity="#fff" d="m6.938 0.5a0.43 0.43 0 0 0 -0.438 0.438v1.25a5.818 5.818 0 0 0 -1.53 0.656l-0.907-0.906a0.436 0.436 0 0 0 -0.625 0l-1.5 1.5a0.436 0.436 0 0 0 0 0.624l0.906 0.907c-0.285 0.48-0.514 0.976-0.656 1.53h-1.25a0.43 0.43 0 0 0 -0.438 0.438v2.125c0 0.248 0.19 0.438 0.438 0.438h1.25a5.82 5.82 0 0 0 0.656 1.53l-0.907 0.908a0.436 0.436 0 0 0 0 0.625l1.5 1.5c0.176 0.176 0.45 0.176 0.625 0l0.907-0.907c0.48 0.285 0.976 0.514 1.53 0.656v1.25c0 0.25 0.19 0.438 0.437 0.438h2.125a0.43 0.43 0 0 0 0.438 -0.438v-1.25a5.82 5.82 0 0 0 1.53 -0.657l0.907 0.907c0.176 0.175 0.45 0.175 0.625 0l1.5-1.5a0.436 0.436 0 0 0 0 -0.625l-0.906-0.906a5.79 5.79 0 0 0 0.657 -1.531h1.25a0.43 0.43 0 0 0 0.438 -0.438v-2.125a0.43 0.43 0 0 0 -0.438 -0.437h-1.25a5.79 5.79 0 0 0 -0.656 -1.532l0.906-0.906a0.436 0.436 0 0 0 0 -0.625l-1.5-1.5a0.436 0.436 0 0 0 -0.625 0l-0.906 0.906a5.816 5.816 0 0 0 -1.53 -0.656v-1.25a0.43 0.43 0 0 0 -0.438 -0.437zm1.062 4.157a3.344 3.344 0 0 1 0 6.6861 3.344 3.344 0 0 1 0 -6.6861z"/></svg>
```
Comment (MorrisJobke) [inline core/img/actions/settings-white.svg]: This icon is somehow black now (instead of the white version) - it's the top right menu icon.

**Review by MorrisJobke — CHANGES_REQUESTED** [review]
Beside the one icon it looks good 👍

File: core/img/actions/settings-white.svg (hunk 1 above)
Comment (pixelipo) [reply core/img/actions/settings-white.svg]: Ooops :scream: fixed now :angel:

**Review by MorrisJobke — APPROVED** [review]

**Comment by MorrisJobke** [comment]
@nextcloud/designers please review

**Review by jancborchardt — APPROVED** [review]
Looking good! Thanks a lot @pixelipo

**Comment by jancborchardt** [comment]
I just have two follow-up questions:
- I saw the strange icons user-plus and user-times before. Haven’t found where/if they are used. Can we delete them?
- What do we need the PNG icons for? They used to be IE8 fallbacks, but we don’t need that anymore? Is it for notification emails?

**Comment by pixelipo** [comment]
@jancborchardt I noticed those as well and haven't touched them. I also couldn't find them used anywhere else and frankly, they shouldn't be, since they are are completely different to `user.svg`. Perhaps in some chat app? @nextcloud/chat 

PNG files are called in places like https://github.com/nextcloud/server/blob/master/apps/comments/lib/Activity/Provider.php#L93
I would love to get rid of them (if possible), and I'll create a separate issue/PR to address that.

**Comment by sualko** [comment]
> Perhaps in some chat app?

No, not that I know of.

**Comment by jancborchardt** [comment]
Ok then feel free to also remove the user-plus and user-times icon in a PR of yours @pixelipo. :)

**Comment by skjnldsv** [comment]
Awesome work here @pixelipo !!

**Comment by MorrisJobke** [comment]
> PNG files are called in places like https://github.com/nextcloud/server/blob/master/apps/comments/lib/Activity/Provider.php#L93
> I would love to get rid of them (if possible), and I'll create a separate issue/PR to address that. - P.S. added it to my TODO in #5157

No - they are needed there. For example on iOS the activity needs to serve PNGs instead of SVGs and some emails also use that AFAIK. cc @nickvergessen

**Pull request merged**